--- ./src/common/MessageHandler.h.orig	2023-09-02 20:20:07.510967610 +0700
+++ ./src/common/MessageHandler.h	2023-09-02 20:37:08.026233588 +0700
@@ -70,24 +70,6 @@
 #endif
 
     static void standardLogger(QtMsgType type, const QString &msg) {
-        static QFile file(QStringLiteral(LOG_FILE));
-
-        // Try to open the log file if we're not outputting to a terminal
-        if (!file.isOpen() && !isatty(STDERR_FILENO)) {
-            if (!file.open(QFile::Append | QFile::WriteOnly))
-                file.open(QFile::Truncate | QFile::WriteOnly);
-
-            // If we can't open the file, create it in a writable location
-            // It will look spmething like ~/.local/share/$appname/sddm.log
-            // or for the sddm user /var/lib/sddm/.local/share/$appname/sddm.log
-            if (!file.isOpen()) {
-                QDir().mkpath(QStandardPaths::writableLocation(QStandardPaths::AppDataLocation));
-                file.setFileName(QStandardPaths::writableLocation(QStandardPaths::AppDataLocation) + QLatin1String("/sddm.log"));
-                if (!file.open(QFile::Append | QFile::WriteOnly))
-                    file.open(QFile::Truncate | QFile::WriteOnly);
-            }
-        }
-
         // create timestamp
         QString timestamp = QDateTime::currentDateTime().toString(QStringLiteral("hh:mm:ss.zzz"));
 
@@ -111,13 +93,8 @@
         QString logMessage = QStringLiteral("[%1] %2 %3\n").arg(timestamp).arg(logPriority).arg(msg);
 
         // log message
-        if (file.isOpen()) {
-            file.write(logMessage.toLocal8Bit());
-            file.flush();
-        } else {
-            fputs(qPrintable(logMessage), stderr);
-            fflush(stderr);
-        }
+        fputs(qPrintable(logMessage), stderr);
+        fflush(stderr);
     }
 
     static void messageHandler(QtMsgType type, const QMessageLogContext &context, const QString &prefix, const QString &msg) {
